tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Provides a library to deal with the CoCo KBS
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kubernetes_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# shellcheck disable=1091
+source "${kubernetes_dir}/../../gha-run-k8s-common.sh"
+
+# Where the kbs sources will be cloned
+readonly COCO_KBS_DIR="/tmp/kbs"
+# The k8s namespace where the kbs service is deployed
+readonly KBS_NS="coco-tenant"
+# The kbs service name
+readonly KBS_SVC_NAME="kbs"
+
+# Delete the kbs on Kubernetes
+#
+# Note: assume the kbs sources were cloned to $COCO_KBS_DIR
+#
+function kbs_k8s_delete() {
+	pushd "$COCO_KBS_DIR"
+	kubectl delete -k kbs/config/kubernetes/overlays
+	popd
+}
+
+# Deploy the kbs on Kubernetes
+#
+function kbs_k8s_deploy() {
+	local image
+	local image_tag
+	local ingress="${1:-}"
+	local repo
+	local kbs_ip
+	local kbs_port
+	local version
+
+	# yq is needed by get_from_kata_deps
+	ensure_yq
+
+	# Read from versions.yaml
+	repo=$(get_from_kata_deps "externals.coco-kbs.url")
+	version=$(get_from_kata_deps "externals.coco-kbs.version")
+	image=$(get_from_kata_deps "externals.coco-kbs.image")
+	image_tag=$(get_from_kata_deps "externals.coco-kbs.image_tag")
+
+	if [ -d "$COCO_KBS_DIR" ]; then
+		rm -rf "$COCO_KBS_DIR"
+	fi
+
+	echo "::group::Clone the kbs sources"
+	git clone --depth 1 "${repo}" "$COCO_KBS_DIR"
+	pushd "$COCO_KBS_DIR"
+	git fetch --depth=1 origin "${version}"
+	git checkout FETCH_HEAD -b kbs_$$
+	echo "::endgroup::"
+
+	pushd kbs/config/kubernetes/
+
+	# Tests should fill kbs resources later, however, the deployment
+	# expects at least one secret served at install time.
+	echo "somesecret" > overlays/key.bin
+
+	echo "::group::Update the kbs container image"
+	install_kustomize
+	pushd base
+	kustomize edit set image "kbs-container-image=${image}:${image_tag}"
+	popd
+	echo "::endgroup::"
+
+	echo "::group::Deploy the KBS"
+	./deploy-kbs.sh
+	popd
+	popd
+
+	if ! kubectl -n "$KBS_NS" wait --for=condition=Ready --timeout=2m \
+		-l app=kbs pod; then
```

Comment:
I got `error: no matching resources found` here as the pod was not yet defined when we reached this. We should perhaps use `waitfor` for the pod definition first to avoid random failures in this section

Reply:
This isn't actually odd, you've found a bug. The kbs pod is created out of a daemonset (ds), so it is perfectly fine it not exist in case the ds couldn't schedule pods. I will switch to the `waitfor` as you suggested. Thanks @ldoktor !